packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1073,6 +1073,10 @@ class TabBarView extends StatefulWidget {
   /// Creates a page view with one child per tab.
   ///
   /// The length of [children] must be the same as the [controller]'s length.
+  ///
+  /// Widgets in [children] that want to stay alive via [KeepAlive] or [AutomaticKeepAlive]
+  /// should be keyed with a [Key] because the TabBarView may re-order its
+  /// children to archive certain scroll effects.
```

Comment:
achieve, not archive

Reply:
will keying them work? i would expect that we wrap the children in our own widgets (like AutomaticKeepAlive or RepaintBoundary). Shouldn't we just automatically key them to their index?

Reply:
@goderbauer and I have discussed whether we should auto key them. We end up not doing it. Should we?

Reply:
added the KeyedSubtree wrap